nix/libff.nix

```diff
@@ -12,7 +12,7 @@ stdenv.mkDerivation rec {
     fetchSubmodules = true;
   };
 
-  cmakeFlags = [ "-DWITH_PROCPS=Off" ];
+  cmakeFlags = [ "-DWITH_PROCPS=OFF -DBUILD_SHARED_LIBS=OFF" ];
```

Comment:
hmm, I've checked and this flag doesn't seem to influence anything.

Reply:
Which flag, actually? I did this change out of caution, and it may not be needed.

The `Off` almost certainly should be changed to `OFF`, not only because of convention, but because of how `if` [works in cmake](https://cmake.org/cmake/help/latest/command/if.html#basic-expressions).

Regarding `-DBUILD_SHARED_LIBS=OFF` -- you re-built it without this change, and the built `hevm` binary is static? Often, in my experience, `-DBUILD_SHARED_LIBS=OFF` helps build static libraries & binaries.

Reply:
I meant `-DBUILD_SHARED_LIBS=OFF`

Reply:
I have just checked, (took about 15 mins of build & preparation time, due to `nix-build` restarting from scratch, and I wanted to be sure), but this statement seems incorrect. Removing `-DBUILD_SHARED_LIBS=OFF` makes this reproducibly incorrect on my end, we get a build error.

In case I am wrong, please post full build log here, so I can check.  To get full build long, apply the DIFF in the 1st comment so Main compiles, run `nix-build`, check the `nix log`, take the option set as per `configureFlags:`, remove the `$...` parts, run it as `cabal build ...` under `nix shell` and don't forget to change `--verbose` to `--verbose=3`. Then post the whole build log here. I believe you should get the same build error I am getting, about linking errors.